Kudu.Services/Diagnostics/LinuxProcessController.cs

```diff
@@ -107,5 +157,69 @@ public IActionResult GetEnvironments(int id, string filter)
 
             return Ok(ArmUtils.AddEnvelopeOnArmRequest(new ProcessEnvironmentInfo(filter, envs), Request));
         }
+
+        private Task ExecuteIfDotnetMonitorEnabled(Func<string, Task> action)
+        {
+            if (DotNetHelper.IsDotNetMonitorEnabled())
+            {
+                var dotnetMonitorAddress = GetDotNetMonitorAddress();
+                if (!string.IsNullOrWhiteSpace(dotnetMonitorAddress))
+                {
+                    return action(dotnetMonitorAddress);
+                }
+
+                Response.StatusCode = (int)HttpStatusCode.NotFound;
+                return Task.FromResult(Response.Body.WriteAsync(Encoding.UTF8.GetBytes(DOTNETMONITORNOTCONFIGURED)));
+            }
+
+            Response.StatusCode = (int)HttpStatusCode.BadRequest;
+            return Task.FromResult(Response.Body.WriteAsync(Encoding.UTF8.GetBytes(ERRORMSG)));
+        }
+
+        private string GetDotNetMonitorAddress()
+        {
+            if (OSDetector.IsOnWindows())
+            {
+                return "http://localhost:52323";
+            }
+
+            var dotnetMonitorAddress = _cache.GetOrCreate(DotNetMonitorAddressCacheKey, entry =>
+            {
+                entry.AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(2);
+                var ipAddress = GetIpAddress();
+                if (!string.IsNullOrWhiteSpace(ipAddress))
+                {
+                    return $"http://{ipAddress}:{dotnetMonitorPort}";
+                }
+                return string.Empty;
+            });
+
+            return dotnetMonitorAddress;
+        }
+
+        private string GetIpAddress()
+        {
+            try
+            {
+                string ipAddress = System.IO.File.ReadAllText("/appsvctmp/ipaddr_" + Environment.GetEnvironmentVariable("WEBSITE_ROLE_INSTANCE_ID"));
```

Comment:
Use `Constants` file for constant env var name/path

Reply:
done!